### ISSUE TITLE
Job remains in a wait list if fails in the scheduled tasks 

### ISSUE THREAD

**Issue by Vitaliy-1** [pkp/pkp-lib#8032]
**Describe the bug**
The issue appeared during debugging StatisticsReport scheduled tasks where editors are notified about editorial statistics. If a job fails, it remains in the `jobs` table and no actions are taken further. If other jobs are added to the queue, these failed jobs are ignored. 

I'd expect a correspondent record in a `failed_jobs` table with an error message and failed job ID(s); remove job from the queue by default or execute other actions, like another attempt if it is specified in job/queue options. 

**To Reproduce**
Dispatch a job as a scheduled task, in my case, it was `PKP\Jobs\Notifications\StatisticsReportMail` job dispatched in `PKP\task\StatisticsReport`, which failed to send an email because of a missing import. 

**What application are you using?**
 OJS main branch

**Comment by Vitaliy-1** [comment]
Not in my case, unless there is a long cooldown between attempts.

**Comment by NateWr** [comment]
In my experience, it would try all jobs once before retrying a job a second time. But otherwise there was no cooldown period.

**Comment by asmecher** [comment]
@NateWr, do you have an example of an invalid `use` statement that could make this reproducible?

**Comment by NateWr** [comment]
I think any invalid `use` statement at the top of `PKP\Jobs\Notifications\StatisticsReportMail` would do the trick. So:

```php
use PKP\does\not\Exist;
```

**Comment by touhidurabir** [comment]
@NateWr @Vitaliy-1 I could not find the job `PKP\Jobs\Notifications\StatisticsReportMail` in the `main` branch(probably removed) but I tried to reproduce it in the `PKP\Jobs\Email\EditorialReminder` . the impact follows as
1. if the invalid `use` statement has no impact on the job execution, the job runs just as fine . 
2. if the invalid `use` statement has impact on the job class that is that `use` statement class/trait used by the job, it throws exception and the job goes right into the `failed_jobs` table . 

so basically I could not reproduce this behaviour. Though I dispatched the job with the new improved implementation through cron or schedule task, see the PR details at [pkp/pkp-lib#7105](https://github.com/pkp/pkp-lib/issues/7105#issuecomment-1205015287) . I even tried it through with our current implementation that is running jobs as the end of application life cycle through a `register_shutdown_function` and the job got executed successfully . Perhaps there was something else wrong with the `PKP\Jobs\Notifications\StatisticsReportMail` that was causing the issue ?
